### PR TITLE
ext: remove invalid Jenkins job URL

### PR DESCRIPTION
**Pull request by jirikuncar**
## Description

Removes invalid Jenkins URL from CI env detection.

## Checklist
- [ ] Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html)
- [ ] Tests provided; and/or
- [ ] Description of manual testing performed and explanation is included in the code and/or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by drodriguezhdez — APPROVED** [review]
LGTM 👍

File: ddtrace/ext/ci.py [hunk 1]
```diff
@@ -222,7 +222,6 @@ def extract_jenkins(env):
         git.COMMIT_SHA: env.get("GIT_COMMIT"),
         git.REPOSITORY_URL: env.get("GIT_URL"),
         git.TAG: tag,
-        JOB_URL: env.get("JOB_URL"),
```
Comment (Kyle-Verhoog) [inline ddtrace/ext/ci.py:225]: Was this field only being used for jenkins?

Comment (jirikuncar) [reply ddtrace/ext/ci.py:225]: There is no easy and correct way to generate `JOB_URL` for Jenkins.

Comment (Kyle-Verhoog) [reply ddtrace/ext/ci.py:225]: Ah! I missed that this was under `extract_jenkins` -- silly comment haha

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**